Source/CarthageKit/VersionFile.swift

```diff
@@ -237,6 +237,72 @@ struct VersionFile: Codable {
 	}
 }
 
+/// Creates a version file for the current project in the
+/// Carthage/Build directory which associates its commitish with
+/// the hashes (e.g. SHA256) of the built frameworks for each platform
+/// in order to allow those frameworks to be skipped in future builds.
+///
+/// Derives the current project name from `git remote get-url origin`
+///
+/// Returns a signal that succeeds once the file has been created.
+public func createVersionFileForCurrentProject(
+	platforms: Set<Platform>,
+	buildProducts: [URL],
+	rootDirectoryURL: URL
+) -> SignalProducer<(), CarthageError> {
+
+	// List all remotes known for this repository
+	let allRemotes = launchGitTask(["remote"])
+		.flatMap(.concat) { $0.linesProducer }
+		.map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
+
+	// For each remote, get the `get-url`
+	let allRemoteURLs = allRemotes
+		.flatMap(.concat) { launchGitTask(["remote", "get-url", $0]) }
+		.map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
+		.collect()
```

Comment:
You could probably use `git remote -v` to get the URLs instead.

Reply:
ah, nice. definitely.

Reply:
Done